### ISSUE TITLE
Rounding

### ISSUE THREAD

**Issue by wdavies973** [a14n/dart-decimal#37]
In Java, you can apply a `MathContext` to set precision and rounding for a `BigDecimal`.

An example of this is:
```
MathContext mc = MathContext.DECIMAL32; // sets precision to 7, and rounding to HALF_EVEN
Math.sin(Math.toRadians(parameters.get(0).doubleValue()))
```

With Java, this outputs `0.5`, with `Decimal`, this outputs `0.49999999999999994` (technically correct).

Could you add a function called "toStringAsPrecision(int precision, RoundType roundType)` that will round the decimal?

It would only need to support `HALF_EVEN` rounding, which rounds to the nearest neighbor, unless they are equidistant, in which case it will round to the even neighbor.

Here's precise documentation on how the rounding works (it does not round to the nearest integer!):
```
/**
         * Rounding mode to round towards the {@literal "nearest neighbor"}
         * unless both neighbors are equidistant, in which case, round
         * towards the even neighbor.  Behaves as for
         * {@code RoundingMode.HALF_UP} if the digit to the left of the
         * discarded fraction is odd; behaves as for
         * {@code RoundingMode.HALF_DOWN} if it's even.  Note that this
         * is the rounding mode that statistically minimizes cumulative
         * error when applied repeatedly over a sequence of calculations.
         * It is sometimes known as {@literal "Banker's rounding,"} and is
         * chiefly used in the USA.  This rounding mode is analogous to
         * the rounding policy used for {@code float} and {@code double}
         * arithmetic in Java.
         *
         *<p>Example:
         *<table class="striped">
         * <caption>Rounding mode HALF_EVEN Examples</caption>
         *<thead>
         *<tr style="vertical-align:top"><th scope="col">Input Number</th>
         *    <th scope="col">Input rounded to one digit<br> with {@code HALF_EVEN} rounding
         *</thead>
         *</table>
         */
```

In short, Banker's rounding is applied, with `precision/scale` taken into account.

**Comment by wdavies973** [comment]
Nevermind, found a solution for this.

**Comment by luissalgadofreire** [comment]
Hi, @widavies. May I ask what solution did you find?

**Comment by marinat** [comment]
@widavies hi, pls share solution
